homeassistant/components/script/__init__.py

```diff
@@ -401,7 +401,12 @@ async def _async_run(self, variables, context):
             # Prepare tracing the execution of the script's sequence
             script_trace.set_trace(trace_get())
             with trace_path("sequence"):
-                return await self.script.async_run(variables, context)
+                states = self.hass.states.get(self.entity_id)
+                this = states.as_dict() if states else {"entity_id": self.entity_id}
```

Comment:
Same comment as above, when can this be None?

Reply:
No, but lint complained about states since the return of states.get can be None.